libcaf_core/caf/detail/sync_ring_buffer.test.cpp

```diff
@@ -29,7 +29,7 @@ std::vector<int> consumer(int_buffer& buf, size_t num) {
 }
 
 void producer(int_buffer& buf, int first, int last) {
-  for (auto i = first; i != last; ++i)
+  for (auto i = first; i != last; ++i) // NOLINT(bugprone-use-after-move)
     buf.push_back(std::move(i));
```

Comment:
These are integers, there's no point in moving them anyways.

```suggestion
  for (auto i = first; i != last; ++i)
    buf.push_back(i);
```

Reply:
That was my first thought as well but `sync_ring_buffer` only has a rvalue `push_back` so this wouldn't compile...

Reply:
Then why not just give it a temporary when it asks for it via `buf.push_back(int{i});`?